components/forms/Register.tsx

```diff
@@ -0,0 +1,229 @@
+  setAuthType,
+}: {
+  setServerError: Dispatch<SetStateAction<string | null>>;
+  setAuthType: Dispatch<SetStateAction<'login' | 'register'>>;
+}) {
+  const supabase = useSupabaseClient<Database>();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<CreateUserInput>({
+    resolver: zodResolver(createUserSchema),
+  });
+
+  const onSubmit = async (formData: CreateUserInput) => {
+    const { data, error } = await supabase.auth.signUp({
+      email: formData.email,
+      password: formData.password,
+      options: {
+        data: {
+          username: formData.username,
+        },
+      },
+    });
+    if (error) {
+      setServerError(error.message);
+      setTimeout(() => {
+        setServerError(null);
+      }, 7000);
+    }
+    if (data && !error) {
+      setAuthType('login');
+    }
+  };
+
+  return (
+    <form
+      className="flex flex-col justify-evenly mt-5 relative"
+      onSubmit={handleSubmit(onSubmit)}
+    >
+      <div className="relative">
+        <div className={`${errors.username ? styles.iconError : styles.icon} `}>
+          <svg
+            width="18"
+            height="18"
+            viewBox="0 0 32 38"
+            fill="none"
+            xmlns="http://www.w3.org/2000/svg"
+          >
+            <path
+              d="M16 37.3333L10.6667 32H0V0H32V32H21.3333L16 37.3333ZM16 17.7778C17.7185 17.7778 19.1852 17.1704 20.4 15.9556C21.6148 14.7407 22.2222 13.2741 22.2222 11.5556C22.2222 9.83704 21.6148 8.37037 20.4 7.15556C19.1852 5.94074 17.7185 5.33333 16 5.33333C14.2815 5.33333 12.8148 5.94074 11.6 7.15556C10.3852 8.37037 9.77778 9.83704 9.77778 11.5556C9.77778 13.2741 10.3852 14.7407 11.6 15.9556C12.8148 17.1704 14.2815 17.7778 16 17.7778ZM16 32.3556L19.9111 28.4444H28.4444V26.4C26.8444 24.8296 24.9855 23.5923 22.8676 22.688C20.7484 21.7849 18.4593 21.3333 16 21.3333C13.5407 21.3333 11.2521 21.7849 9.13422 22.688C7.01511 23.5923 5.15556 24.8296 3.55556 26.4V28.4444H12.0889L16 32.3556Z"
+              fill="#FBFBFB"
+            />
+          </svg>
+        </div>
+        <input
+          type="text"
+          className={`form-control
+                w-full
+                py-2 pl-6
+                          self-start
+                          text-base
+                          font-normal
+                          placeholder:text-white
+                          placeholder:opacity-60   
+                          border-2 border-solid border-blueGrey-600
+                          rounded-2xl
+                          transition
+                          ease-in-out
+                          focus:outline-frost-50
+                          m-0 focus:outline-none  bg-inherit flex-1  ${styles.input}`}
+          placeholder="Enter Username"
+          {...register('username')}
+        ></input>
+        {errors.username && (
+          <span className="text-red-700 mt-1 text-sm font-bold">
+            {errors.username?.message}
```

Comment:
Minor thing but since we conditionally render this block when `errors.username` has a value assigned, this null coalescence operator is redundant
```suggestion
            {errors.username.message}
```